src/common/strlib.h

```diff
@@ -70,6 +70,19 @@
 #endif
 #endif
 
+/**
+ * Workaround for gcc issues 80354, 89312, that make it difficult to ignore
+ * -Wformat-truncation warnings in those rare cases where truncation is
+ * irrelevant to the program's correctness.
+ *
+ * @remark Do not use unless you're sure about the implications of ignoring
+ * this warning.
+ */
+#define snprintf_ignore_truncation(str, size, format, ...) do { \
```

Comment:
pragma with ignore -Wformat-truncation not works?
i mean in same was as in enum switch for example.

Reply:
Good idea, it worked. I removed the macro and replaced it with the pragmas in the few places where it was used (I didn't like the macro since it'd make it too tempting to ignore possible real issues)